common/persistence/domainReplicationQueue.go

```diff
@@ -45,25 +48,24 @@ func NewDomainReplicationQueue(
 	logger log.Logger,
 ) DomainReplicationQueue {
 	return &domainReplicationQueueImpl{
-		queue:               queue,
-		clusterName:         clusterName,
-		metricsClient:       metricsClient,
-		logger:              logger,
-		encoder:             codec.NewThriftRWEncoder(),
-		ackNotificationChan: make(chan bool),
-		done:                make(chan bool),
+		queue:           queue,
+		clusterName:     clusterName,
+		metricsClient:   metricsClient,
+		logger:          logger,
+		encoder:         codec.NewThriftRWEncoder(),
+		lastCleanupTime: time.Now().UTC(),
```

Comment:
The requirement to call UTC() everywhere this field is accessed seems error prone and a bug waiting to happen. How about we have the API DeleteMessagesBefore(t time.Time) call t.UTC() so that the callers don't have to worry about this ?

Reply:
With the origin approach, I don't need this time source